### PR TITLE
feat: agent internal collector

### PR DESCRIPTION
**Pull request by mathnogueira**
This PR allows the agent to start a HTTP and gRPC server to receive traces from a local application. This will be used by the local `dev mode` we will introduce.

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

### CODE REVIEW

File: agent/collector/http_server.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	return s.parseProtoBuf(protoBody)
+}
+
+func (s httpServer) parseBody(reqBody io.ReadCloser, contentType string) (*pb.ExportTraceServiceRequest, error) {
+	var body []byte
+	if b, err := io.ReadAll(reqBody); err == nil {
+		body = b
+	} else {
+		return nil, err
+	}
```
Comment (danielbdias) [inline agent/collector/http_server.go:114]: Is there a reason to use this `if/else`, or can we simplify it using an early return, like this?
```suggestion
	var body []byte
	body, err := io.ReadAll(reqBody)
	
	if err != nil {
		return nil, err
	}
```

File: agent/collector/http_server.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+		w:           w,
+		contentType: contentType,
+	}
+}
+
+func (r httpResponse) send(statusCode int, message proto.Message) error {
+	body, err := r.paseResponseBody(message)
+	if err != nil {
+		fmt.Println("Could not attach body to response", err.Error())
+		return err
```
Comment (danielbdias) [inline agent/collector/http_server.go:143]: Does it make sense to wrap this error with this message too?
```suggestion
		fmt.Println("Could not attach body to response", err.Error())
		return fmt.Errorf("could not attach body to response: %w", err)
```

**Review by danielbdias — APPROVED** [review]

File: server/otlp/ingester.go [hunk 3]
```diff
@@ -16,21 +16,32 @@ import (
 	v1 "go.opentelemetry.io/proto/otlp/trace/v1"
 )
 
+type Ingester interface {
```
Comment (schoren) [inline server/otlp/ingester.go:19]: is this change required? looks like no other package is referencing the interface. maybe I'm missing something though

File: agent/collector/ingester.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	"time"
+
+	"github.com/kubeshop/tracetest/server/otlp"
+	pb "go.opentelemetry.io/proto/otlp/collector/trace/v1"
+	v1 "go.opentelemetry.io/proto/otlp/trace/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type ingester interface {
```
Comment (schoren) [inline agent/collector/ingester.go:17]: this interface also seems unused, maybe it can be removed?

**Review by schoren — APPROVED** [review]
looks awesome! If i'm not mistaken, the server ingestor interface can be left unexported and the agent's ingestor interface is not used, but I might be missing something

File: agent/collector/ingester.go (hunk 4 above)
Comment (mathnogueira) [reply agent/collector/ingester.go]: changed it to `stoppable` because we might have to stop the ingester.

File: server/otlp/ingester.go (hunk 3 above)
Comment (mathnogueira) [reply server/otlp/ingester.go:19]: after I deleted the `collector.ingester` interface, we needed this to be exported.
